### DISCUSSION TITLE
Help with some reactive code

### DISCUSSION THREAD

**Discussion by dkoci** [quarkusio/quarkus#36381] [Q&A]
Hi.
I am learning the non-blocking reactive approach and I need some help. For each element in the `Uni<List<Item>>`, I want to call an external REST API that returns the status for each Item. Then, I want to update the status in the database for each Item. How would I do this correctly?

Thank you

**Comment by ennishol** [comment]
Map uni to multi then you can do whatever you want with each item in uni

`Uni<List<T>> results = list();
        return (Multi<T>) results.toMulti().flatMap(list -> {
            return Multi.createFrom().iterable(list);
        });`
